diff --git a/node_modules/gatsby-theme-mdx-deck/src/components/import-slides.js b/node_modules/gatsby-theme-mdx-deck/src/components/import-slides.js
new file mode 100644
index 0000000..84151f0
--- /dev/null
+++ b/node_modules/gatsby-theme-mdx-deck/src/components/import-slides.js
@@ -0,0 +1,5 @@
+export const ImportSlides = props => {
+    return false
+  }
+  
+  export default ImportSlides
\ No newline at end of file
diff --git a/node_modules/gatsby-theme-mdx-deck/src/components/presenter.js b/node_modules/gatsby-theme-mdx-deck/src/components/presenter.js
index 7d33382..502e8bf 100644
--- a/node_modules/gatsby-theme-mdx-deck/src/components/presenter.js
+++ b/node_modules/gatsby-theme-mdx-deck/src/components/presenter.js
@@ -41,7 +41,7 @@ export const Presenter = ({ slides, children }) => {
             p: 3,
           }}>
           <Zoom ratio={4 / 3} zoom={1 / 4}>
-            <Slide slide={next} preview />
+            <Slide slide={next} index={context.index + 1} preview />
           </Zoom>
           {notes && (
             <div
diff --git a/node_modules/gatsby-theme-mdx-deck/src/components/slide-list.js b/node_modules/gatsby-theme-mdx-deck/src/components/slide-list.js
index a8bdcd8..d647f1c 100644
--- a/node_modules/gatsby-theme-mdx-deck/src/components/slide-list.js
+++ b/node_modules/gatsby-theme-mdx-deck/src/components/slide-list.js
@@ -50,7 +50,7 @@ export const SlideList = ({
             outline: index === i ? `4px solid cyan` : null,
           }}>
           <Zoom zoom={zoom} ratio={ratio}>
-            <Slide slide={slide} preview />
+            <Slide slide={slide} index={i} preview />
           </Zoom>
         </div>
       ))}
diff --git a/node_modules/gatsby-theme-mdx-deck/src/index.js b/node_modules/gatsby-theme-mdx-deck/src/index.js
index 61e519f..642d493 100644
--- a/node_modules/gatsby-theme-mdx-deck/src/index.js
+++ b/node_modules/gatsby-theme-mdx-deck/src/index.js
@@ -12,6 +12,7 @@ export { Slide } from './components/slide'
 export { Zoom } from './components/zoom'
 export { Embed } from './components/embed'
 export { Image } from './components/image'
+export { ImportSlides } from './components/import-slides'
 export { FullScreenCode } from './components/full-screen-code'
 export { Horizontal } from './components/horizontal'
 export { Invert } from './components/invert'
diff --git a/node_modules/gatsby-theme-mdx-deck/src/split-slides.js b/node_modules/gatsby-theme-mdx-deck/src/split-slides.js
index 6f1bcd8..b530a10 100644
--- a/node_modules/gatsby-theme-mdx-deck/src/split-slides.js
+++ b/node_modules/gatsby-theme-mdx-deck/src/split-slides.js
@@ -8,7 +8,23 @@ export default props => {
     props: {},
     children: [],
   }
-  arr.forEach((child, i) => {
+  
+  const flattenedArray = arr.reduce((acc, child) => {
+    if (
+      child.props.mdxType === 'ImportSlides' &&
+      typeof child.props.src === 'function'
+    ) {
+      const innerSlides = child.props.src({});
+      if (innerSlides.props.children) {
+        acc.push(...React.Children.toArray(innerSlides.props.children))
+      }
+      return acc
+    }
+    acc.push(child)
+    return acc
+  }, [])
+  
+  flattenedArray.forEach((child, i) => {
     const {
       originalType,
       mdxType,
@@ -25,12 +41,12 @@ export default props => {
   })
   let previousSplit = 0
   splits.forEach(i => {
-    const children = [...arr.slice(previousSplit, i)]
+    const children = [...flattenedArray.slice(previousSplit, i)]
     slides.push(children)
     previousSplit = i + 1
   })
 
-  slides.push([...arr.slice(previousSplit)])
+  slides.push([...flattenedArray.slice(previousSplit)])
 
   slides.head.children = React.Children.toArray(slides.head.children).map(
     (child, i) => {
